lib/personnummer.dart

```diff
@@ -80,125 +121,113 @@ class Personnummer {
       }
     }
 
-    map['century'] = century;
-    map['year'] = year;
-    map['month'] = month;
-    map['day'] = day;
-    map['sep'] = sep;
-    map['nm'] = nm;
-    map['check'] = check;
+    this.century = century;
+    this.fullYear = century + year;
+    this.year = year;
+    this.month = month;
+    this.day = day;
+    this.sep = sep;
+    this.num = nm;
+    this.check = check;
+
+    int ageDay = int.parse(day);
+    if (ageDay >= 61 && ageDay <= 91) {
+      ageDay -= 60;
+    }
 
-    return map;
-  }
+    DateTime u =
+        new DateTime(int.parse(century + year), int.parse(month), ageDay);
+    DateTime dt = dateTimeNow == null ? DateTime.now() : dateTimeNow;
 
-  /// Get the age from a personnummer.
-  static int getAge(dynamic input, [bool includeCoordinationNumber = true]) {
-    if (!valid(input, includeCoordinationNumber)) {
-      throw new PersonnummerException();
-    }
+    this.age =
+        (dt.difference(u).inMilliseconds / 3.15576e+10).floor().toString();
 
-    HashMap parts = getParts(input);
-    if (parts.isEmpty) {
+    if (!this._valid()) {
       throw new PersonnummerException();
     }
+  }
 
-    int day = int.parse(parts['day']);
-    if (day >= 61 && day <= 91) {
-      day -= 60;
-    }
-
-    DateTime u = new DateTime(int.parse(parts['century'] + parts['year']),
-        int.parse(parts['month']), day);
-    DateTime dt = dateTimeNow == null ? DateTime.now() : dateTimeNow;
-
-    return (dt.difference(u).inMilliseconds / 3.15576e+10).floor();
+  /// Test year, month and day as date and see if it's the same.
+  /// Returns `true` if it's the same.
+  bool _testDate(int year, int month, int day) {
```

Comment:
Will this not always return true unless the `new DateTime` throws an error? :D

Reply:
Or maybe it does not throw an error and will overflow to next month if a coordination number?

Reply:
This is mostly used after the parser to actual test that the input is a real date or not. The parser don't test date yet. I'll leave as it right now :)